### PR TITLE
Compile rides as C++

### PR DESCRIPTION
**Pull request by janisozaur**
What could possibly go wrong?

### CODE REVIEW

**Comment by IntelOrca** [comment]
image

obviously...

**Comment by janisozaur** [comment]
@Gymnasiast @marijnvdwerf @LRFLEW can you apply Xcode fix? That will have at least Travis green.

**Comment by Gymnasiast** [comment]
Done the Xcode fix, but the .h files still need to be renamed.

**Comment by marijnvdwerf** [comment]
@Gymnasiast I was attempting this too (didn't see you were also at it), and I got linker issues regarding the paint functions. Did you have those as well?

**Comment by Gymnasiast** [comment]
@marijnvdwerf No, I compiled it and started it up. I had no issues with that.

**Comment by janisozaur** [comment]
> the .h files still need to be renamed.

Oh, I didn't notice them. Good spot.

**Comment by Gymnasiast** [comment]
I can do it, and fix the Xcode with it.

**Comment by janisozaur** [comment]
@Gymnasiast please do

**Comment by Gymnasiast** [comment]
Fixed SpiralSlide.cpp for VS, but MiniGolf.cpp is still littered with
`ride\gentle\MiniGolf.cpp(765): error C4576: a parenthesized type followed by an initializer list is a non-standard explicit type conversion syntax [C:\projects\openrct2-ject9\src\openrct2\libopenrct2.vcxproj]`

**Comment by janisozaur** [comment]
If it's really only `MiniGolf.cpp` now, it's not as bad as I imagined.

**Comment by Gymnasiast** [comment]
Looks like it. And it shouldn't be that difficult to fix, by the look of it.

**Comment by janisozaur** [comment]
@Gymnasiast if you have the time now, you can take a stab at it. If not, I will probably change it in the evening.

For `mini_golf_peep_animation_lengths` I'd first try if it can be changed to `size_t`.

**Comment by Gymnasiast** [comment]
@janisozaur That is just a warning though. I have checked, it is used in a comparison, not saved into a fixed-width structure. In other words, it can safely be changed.

File: src/openrct2/ride/gentle/MiniGolf.cpp [hunk 1]
```diff
@@ -822,8 +832,16 @@ static void paint_mini_golf_hole_c(paint_session * session, uint8 rideIndex, uin
         paint_util_push_tunnel_right(session, height, TUNNEL_10);
     }
 
-    rct_xy16 boundBox       = (direction & 1) ? (rct_xy16){ 26, 32 } : (rct_xy16){ 32, 26 };
-    rct_xy16 boundBoxOffset = (direction & 1) ? (rct_xy16){ 3, 0 } : (rct_xy16){ 0, 3 };
+    if (direction & 1)
+    {
+        rct_xy16 boundBox       = { 26, 32 };
```
Comment (janisozaur) [inline src/openrct2/ride/gentle/MiniGolf.cpp]: you re-declare the variable here

Comment (Gymnasiast) [reply src/openrct2/ride/gentle/MiniGolf.cpp]: Which Xcode, for some reason, decided to allow :'(

**Comment by janisozaur** [comment]
@Gymnasiast https://www.diffchecker.com/3bxq8MPQ

File: src/openrct2/ride/gentle/MiniGolf.cpp [hunk 2]
```diff
@@ -889,8 +908,16 @@ static void paint_mini_golf_hole_d(paint_session * session, uint8 rideIndex, uin
         break;
     }
 
-    rct_xy16 boundBox       = (supportType & 1) ? (rct_xy16){ 26, 32 } : (rct_xy16){ 32, 26 };
-    rct_xy16 boundBoxOffset = (supportType & 1) ? (rct_xy16){ 3, 0 } : (rct_xy16){ 0, 3 };
+    if (direction & 1)
```
Comment (janisozaur) [inline src/openrct2/ride/gentle/MiniGolf.cpp]: `supportType`, not `direction`

File: src/openrct2/ride/gentle/MiniGolf.cpp [hunk 3]
```diff
@@ -972,8 +1000,16 @@ static void paint_mini_golf_hole_e(paint_session * session, uint8 rideIndex, uin
...
```
Comment (janisozaur) [inline src/openrct2/ride/gentle/MiniGolf.cpp]: `supportType`, not `direction`

**Comment by Gymnasiast** [comment]
I pushed a fix for the two errors - can you verify the testpaint output?

**Comment by janisozaur** [comment]
@Gymnasiast much better now: https://travis-ci.org/OpenRCT2/OpenRCT2/jobs/285580527#L998

Same as `develop`.

**Comment by Gymnasiast** [comment]
Alright!
Anything else to check before pressing the big green button?

**Comment by janisozaur** [comment]
That's all, :shipit:

**Comment by janisozaur** [comment]
Woooo!

**Pull request merged**